src/Core/Repositories/SqlServer/CollectionRepository.cs

```diff
@@ -107,7 +107,21 @@ public async Task<ICollection<CollectionDetails>> GetManyByUserIdAsync(Guid user
                 // Return distinct Id results.
                 return results
                     .GroupBy(c => c.Id)
-                    .Select(c => c.First())
+                    .Select(grouping =>
+                    { 
+                        var first = grouping.First();
+                        if (first.HidePasswords)
+                        {
+                            first.HidePasswords = !grouping.Any(c => !c.HidePasswords);
+                        }
+
+                        if (first.ReadOnly)
+                        {
+                            first.ReadOnly = !grouping.Any(c => !c.ReadOnly);
+                        }
+                        
+                        return first;
```

Comment:
This kind of filtering logic (even though some of it already existed here) seems out of place in the repository. It doesn't seem like a huge life to move this logic into `Collection_ReadByUserId.sql`. Maybe we could improve that while in the area and relocate this?